### PR TITLE
Java2

### PR DESCRIPTION
**Pull request by Aayushi1301**
Completed Java Assignment 2

### CODE REVIEW

File: JAVA2/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+out/
```
Comment (ghost) [inline JAVA2/.gitignore]: Out directory also got pushed. Make sure that gitignore is working properly before committing the changes.

File: JAVA2/src/main/Main.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import solution.stringContainsAllLettersOfAlphabet;
+import java.util.Scanner;
+/**
+ * Main class.
+ * Program to calculate whether the input by the user contains all letters from a-z
+ */
+public class Main {
+    public static void main(String[] args) {
+
+        /**
```
Comment (ghost) [inline JAVA2/src/main/Main.java]: doc must be outside the method just before the the declaration

File: JAVA2/src/solution/stringContainsAllLettersOfAlphabet.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package solution;
+import java.util.HashSet;
+import java.util.Set;
+/**
+ * Write a java function that checks if the input string contains all the letters of the
+ * alphabet a-z (case-insensitive).
+ * Time Complexity:O(1)
+ * Space Complexity:O(n) where n is size of input
+ */
+public class stringContainsAllLettersOfAlphabet {
```
Comment (ghost) [inline JAVA2/src/solution/stringContainsAllLettersOfAlphabet.java]: class names must begin with caps

File: JAVA2/src/solution/stringContainsAllLettersOfAlphabet.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    public boolean check(String str)
+    {
+        /**
+         * Checks whether input string contains all letters of an alphabet.
```
Comment (ghost) [inline JAVA2/src/solution/stringContainsAllLettersOfAlphabet.java]: Same applicable for docs here. Must be outside the method

File: JAVA2/src/test/ContainsAllAlphabetTest.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package test;
+import org.junit.Test;
+import solution.stringContainsAllLettersOfAlphabet;
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.assertFalse;
+/**
+ * Test class to test CheckAlphabets class
+ */
+public class ContainsAllAlphabetTest {
```
Comment (ghost) [inline JAVA2/src/test/ContainsAllAlphabetTest.java:9]: need few more test cases.

File: JAVA2/src/solution/stringContainsAllLettersOfAlphabet.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+            for (char c : str.toUpperCase().toCharArray()) {
+                if (Character.isLetter(c))
+                    set.add(c);
+            }
+
+        }
+        return (set.size() == 26);
+    }
+
```
Comment (ghost) [inline JAVA2/src/solution/stringContainsAllLettersOfAlphabet.java]: Please specify the time complexity of your solution too.

**Review by ghost — CHANGES_REQUESTED** [review]
Please make the necessary changes to the code and push the updated changes.

File: JAVA2/src/solution/stringContainsAllLettersOfAlphabet.java (hunk 6 above)
Comment (Aayushi1301) [reply JAVA2/src/solution/stringContainsAllLettersOfAlphabet.java]: I have added it in the beginning of the class stringContainsAllLettersOfAlphabet

File: JAVA2/src/solution/stringContainsAllLettersOfAlphabet.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
+package solution;
+import java.util.HashSet;
+import java.util.Set;
+/**
+ * Write a java function that checks if the input string contains all the letters of the
+ * alphabet a-z (case-insensitive).
+ * Time Complexity:O(1)
```
Comment (ghost) [inline JAVA2/src/solution/stringContainsAllLettersOfAlphabet.java]: Can you verify these values?

**Review by ghost — APPROVED** [review]

**Pull request merged**